SDWebImage/Core/SDImageAWebPCoder.m

```diff
@@ -56,12 +55,11 @@ - (BOOL)canIncrementalDecodeFromData:(NSData *)data {
 }
 
 - (BOOL)canEncodeToFormat:(SDImageFormat)format {
-    switch (format) {
-        case SDImageFormatWebP:
-            // Check WebP encoding compatibility
-            return [self.class canEncodeToFormat:SDImageFormatWebP];
-        default:
-            return NO;
+    if (format == SDImageFormatWebP) {
```

Comment:
Compiler can do optimization based on enum cases number, these changes is no needed.

Reply:
![image](https://github.com/SDWebImage/SDWebImage/assets/7018936/4df2d101-5539-488c-a784-c77ade33b833)

The compiler reports an error when declare these variables as extern.

Reply:
So, is this cause API break (assume user use code like this in switch case?)

If so, why not totally break and use class declarations instead

Reply:
I remembered...In 5.c


This static declaration is written by me to allows "switch case" works

This code is indeed, so seems this is not possible in 5.x